src/FakeItEasy/Core/MethodInfoManager.cs

```diff
@@ -53,7 +53,7 @@ select MakeGeneric(typeMethod, method)).FirstOrDefault()
 
         private static MethodInfo? GetMethodOnInterfaceTypeImplementedByMethod(Type type, MethodInfo method)
         {
-            var reflectedType = method.ReflectedType;
+            var reflectedType = method.ReflectedType!;

```

Comment:
RefectedType can only be null for non-nested types, module-level methods, or dynamic methods.